golos.publication/golos.publication.cpp

```diff
@@ -334,31 +334,58 @@ void publication::unvote(name voter, structures::mssgid message_id) {
     set_vote(voter, message_id, 0);
 }
 
-void publication::payto(name user, eosio::asset quantity, enum_t mode, std::string memo) {
+void publication::payto(std::vector<eosio::token::recipient> recipients, enum_t mode, symbol tokensymbol) {
     require_auth(_self);
-    eosio_assert(quantity.amount >= 0, "LOGIC ERROR! publication::payto: quantity.amount < 0");
-    if(quantity.amount == 0)
+    
+    auto publish_payout = eosio::asset(0, tokensymbol);
+    std::vector<eosio::token::recipient> recipients_payout;
+    std::vector<name> wrong_quantity;
+
+    if (!recipients.size())
         return;
 
-    if(static_cast<payment_t>(mode) == payment_t::TOKEN) {
-        if (token::balance_exist(config::token_name, user, quantity.symbol.code())) {
-            INLINE_ACTION_SENDER(token, payment) (config::token_name, {_self, config::code_name}, {_self, user, quantity, memo});
-        } 
-        else {
-            INLINE_ACTION_SENDER(token, payment) (config::token_name, {_self, config::code_name}, {_self, _self, quantity, memo});
-        }
-    }
-    else if(static_cast<payment_t>(mode) == payment_t::VESTING) {
-        if (golos::vesting::balance_exist(config::vesting_name, user, quantity.symbol.code())) {
-            INLINE_ACTION_SENDER(token, transfer) (config::token_name, {_self, config::code_name},
-            {_self, config::vesting_name, quantity, std::string(config::send_prefix + name{user}.to_string()  + "; " + memo)});
-        } 
-        else {
-            INLINE_ACTION_SENDER(token, payment) (config::token_name, {_self, config::code_name}, {_self, _self, quantity, memo});
+    auto set_payout = [&](payment_t mode) {
+        auto check_balance = [&](payment_t mode, name acc, symbol sym) -> bool {
+            if (mode == payment_t::TOKEN)
+                return token::balance_exist(config::token_name, acc, sym.code());
+            else
+                return golos::vesting::balance_exist(config::vesting_name, acc, sym.code());
+        };
+        
+        for (auto recipient_obj : recipients) {
+            if (recipient_obj.quantity.amount >= 0) {
+                if (check_balance(mode, recipient_obj.to, recipient_obj.quantity.symbol)) {
+                    if (recipient_obj.quantity.amount)
+                        recipients_payout.push_back(recipient_obj);
+                } else {
+                    if (recipient_obj.quantity.amount)
+                        publish_payout.amount += recipient_obj.quantity.amount;
+                }
+            } else {
+                wrong_quantity.push_back(recipient_obj.to);
+            }
         }
-    }
-    else
+    };
+
+    if (static_cast<payment_t>(mode) == payment_t::TOKEN) {
+        set_payout(payment_t::TOKEN); 
```

Comment:
Try to write this code without lambda-function `set_payout`. Using this lambda make code confused.